app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/OnboardingScreen.kt

```diff
@@ -0,0 +1,195 @@
+import com.infomaniak.swisstransfer.ui.images.illus.onboarding.*
+import com.infomaniak.swisstransfer.ui.screen.onboarding.components.AnimatedOnboardingButton
+import com.infomaniak.swisstransfer.ui.theme.Dimens
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewLargeWindow
+import com.infomaniak.swisstransfer.ui.utils.PreviewSmallWindow
+import kotlinx.coroutines.launch
+
+@Composable
+fun OnboardingScreen(goToMainActivity: () -> Unit) {
+    val pagerState = rememberPagerState(pageCount = { Page.entries.size })
+    val isLastPage by remember { derivedStateOf { pagerState.currentPage >= pagerState.pageCount - 1 } }
+    val coroutineScope = rememberCoroutineScope()
+
+    val isHighlighted = Page.entries.associateWith { rememberSaveable { mutableStateOf(false) } }
+
+    // Start the highlighting of the text when the associated page is reached in the HorizontalPager
+    LaunchedEffect(pagerState.currentPage) {
+        val currentPage = Page.entries[pagerState.currentPage]
+        val setIsHighlighted = isHighlighted[currentPage]?.component2()
+        setIsHighlighted?.invoke(true)
+    }
+
+    BackHandler(pagerState.currentPage > 0) {
+        coroutineScope.launch {
+            pagerState.animateScrollToPage(pagerState.currentPage - 1)
+        }
+    }
+
+    val onboardingPages = buildList {
+        Page.entries.forEach { page ->
+            add(page.toOnboardingPage(isHighlighted))
+        }
+    }
+
+    OnboardingScaffold(
+        pagerState = pagerState,
+        onboardingPages = onboardingPages,
+        bottomContent = {
+            BottomContent(
+                modifier = Modifier
+                    .padding(it)
+                    .consumeWindowInsets(it),
+                isLastPage = { isLastPage },
+                startMainActivity = goToMainActivity,
+                goToNextPage = { coroutineScope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) } },
+            )
+        },
+    )
+}
+
+@Composable
+private fun Page.toOnboardingPage(isHighlighted: Map<Page, MutableState<Boolean>>) = OnboardingPage(
+    background = background.image(),
+    illustration = { Illustration(illustration) },
+    text = {
+        TitleAndDescription(
+            page = this,
+            isHighlighted = { isHighlighted[this]?.value ?: false }
+        )
+    }
+)
+
+@Composable
+private fun Illustration(illustration: ThemedImage) {
+    Image(illustration.image(), contentDescription = null)
+}
+
+@Composable
+private fun TitleAndDescription(page: Page, isHighlighted: () -> Boolean) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(page.titleRes),
+            style = SwissTransferTheme.typography.specificLight22,
+        )
+
+        Spacer(modifier = Modifier.height(Margin.Mini))
```

Comment:
I see on the maquettes that this title is supposed to be in the same paragraph as the description. 
This mini padding feels a bit strange, because it's not a real title paddng, but not in the same paragraph as the description either... 
And the text is really big, so a mini Margin feels a bit out of place in my opinion

Reply:
It does indeed, but without it the green part of the highlighted text hides the title because the title is drawn before the description. Apparently there's no way to change the drawing order inside of a Column, it's always top to bottom :/

But I'm not sure what else we can do apart from spacing it a bit